Automattic/jetpack
# Jetpack Blocks: add new Promoted Posts post-publish panel

#### Changes proposed in this Pull Request:

This is a replay of #27591:

- It removes some of the experimental tests to run the Blaze modal inside the editor; this is something we can do in a future iteration.
- It takes advantage of #27923 and #27924.

This PR introduces a new Post publish panel. The new panel will be added in the following conditions:

- After a post has been published, in the post-publish panel.
- on posts, pages, and products.
- On WordPress.com Simple sites and WoA sites.
- On public (launched) sites.
- When the current user is connected to WordPress.com.
- If the user's WordPress.com locale is en or en-gb
- the post must be published, and it should not be password-protected or private.

**The panel is also only registered as Beta for now, so on WordPress.com simple you'll need to be proxied to see the changes, and on other sites you'll need to explicitly enable Jetpack's Beta block bundle to be able to test the change**.

<img width="319" alt="image" src="https://user-images.githubusercontent.com/426388/207642123-80534bfc-2738-4f56-8b2a-34d0d00482b1.png">


**To do**

In a future PR, I'd like to do the following:

- [x] Refactor the panel's analytics implementation; right now it depends on an implementation that only ships with the Jetpack plugin. #27937 and #27940
- [ ] Move the panel to the new Jetpack Promote package, so it can be used in both the Jetpack plugin and the Jetpack Social plugin.

#### Other information:


#### Jetpack product discussion

- 1386-gh-dotcom-forge
- #27591

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start on a WordPress.com simple or WoA site that's connected to WordPress.com.
* Switch to this branch, or use the Jetpack Beta plugin to run this branch.
* Ensure Beta blocks are enabled on your site. If you test on a WoA site, for example, add the following: `add_filter( 'jetpack_blocks_variation', function() { return 'beta'; } );`. If you work locally, you can add 
`define( 'JETPACK_BETA_BLOCKS', true );` in your `docker/wordpress/wp-config.php`.
* Go to Posts > Add New
* In your console, check that the new extension is now available: `'post-publish-promote-post-panel' in Jetpack_Editor_Initial_State.available_blocks`
* Publish a post.
* Check the post-publish sidebar panel and click on the link.
* Try different scenarios (Jetpack site, password-protected post, ...) to see that the panel is hidden when it needs to be hidden.

**Note**: you may notice that when clicking on the link and landing on the Advertising page, at times the page doesn't fully load and the following error appears in the console:
`proxy eror InternalServerError: 500 status code for "GET /sites/143258936/wordads/dsp/api/v1/templates/article/urn:wpcom:post:143258936:2955"`
That is something we're discussing in 1386-gh-dotcom-forge.
